autonomy/cli/scaffold_fsm.py

```diff
@@ -189,6 +190,26 @@ def author(self) -> str:
         """Author"""
         return self.ctx.agent_config.author
 
+    @property
+    def rounds(self) -> Set[str]:
+        """Rounds"""
+        return self.dfa.states
+
+    @property
+    def non_degenerate_rounds(self) -> Set[str]:
+        """Non-degenerate rounds"""
+        return self.rounds - self.dfa.final_states
+
+    @property
+    def behaviours(self) -> Set[str]:
+        """Behaviours"""
+        return {s.replace("Round", "Behaviour") for s in self.non_degenerate_rounds}
```

Comment:
It should be `
```suggestion
        return {s.replace("Round", "Behaviour") for s in self.rounds}
```

And then you can have `non_degenerate_behaviours` property as well. Otherwise there's an inconsistency remaining.

Reply:
we don't have behaviours or payloads for degenerate rounds though

Reply:
We do have degenerate behaviours (they're dynamically generated), otherwise agree, but your naming is inconsistent. I suggest then calling the full set of rounds `all_rounds` and then it's at least consistent (i.e. all rounds/behaviours/payloads properties refer to the set excluding the degenerate ones)

Reply:
addressed [here](https://github.com/valory-xyz/open-autonomy/pull/1301/files#r964951850)